### ISSUE TITLE
white screen (mac Sonoma 14.4.1)

### ISSUE THREAD

**Issue by bsoener** [Gkiokan/ps4-remote-pkg-sender#71]
Since updating Sonoma 14.4.1 on Mac, I only get a white screen. I can no longer open the app.

**Comment by Gkiokan** [comment]
Hi,

thats interesting, apples notes doesn't say anything that could have affected the environment.  
I am myself on 14.2.1 and can't just update my environment in middle of Projects. 
But you should get some kind of Error Window with the exact error happening from the Error Catcher. 

Could you clone the repo and run `npm run watch` just to see what the logs exactly are saying?
nvm should be set to 14.7 or 16 at max if you get lib errors.

**Comment by bsoener** [comment]
Thank you for your answer. Could you please explain to me exactly how to do this?

Thank you very much

**Comment by Gkiokan** [comment]
Are you a terminal user? Make sure you have npm v14 or 16 active before doing this! 
  
`git clone git@github.com:Gkiokan/ps4-remote-pkg-sender.git`. 
`cd ps4-remote-pkg-sender && npm i` . 
`npm run watch` 

Then the application should start and show you also the DevTools which allows you to deep debug the application. There should be a console.log with the specific error probably.

**Comment by bsoener** [comment]
```
root@MacBook-Pro-2 downloads # sudo git clone git@github.com:Gkiokan/ps4-remote-pkg-sender.git
xcrun: error: invalid active developer path (/Library/Developer/CommandLineTools), missing xcrun at: /Library/Developer/CommandLineTools/usr/bin/xcrun
root@MacBook-Pro-2 downloads # 
```

Thank you for your answer. When entering Terminal: git clone I get this message

**Comment by Masamune3210** [comment]
Your toolset is misconfigured. However you set it up, go back and try it again.

**Comment by Gkiokan** [comment]
Meh, this sounds like your env is not configured properly.  

Try to download the source code via the download zip button on GitHub and  
continue the steps after git clone that I've suggested in the terminal (in the folder of the extracted download file ofc).

**Comment by bsoener** [comment]
I packed everything into the folder and executed this command via terminal: npm run watch


```

root@MacBook-Pro-2 ps4-remote-pkg-sender-master #  npm run watch

> PS4RemotePKGSenderV2@2.10.3 watch
> electron-webpack dev

sh: electron-webpack: command not found

```

**Comment by Gkiokan** [comment]
I think it will be easier when I help you getting this running with a remote desktop share session.  
Get in touch with me on Telegram, Discord, Insta or Facebook.

**Comment by tangestani** [comment]
I've got the same issue as the OP and I'm also on Sonoma 14.4.1.

Here's what I get when I run `npm run watch`: [debug.log](https://github.com/Gkiokan/ps4-remote-pkg-sender/files/15342151/debug.log)

Let me know if there's anything else I could do to help narrow down the issue.

**Comment by tangestani** [comment]
And here are the errors I see when connecting to `http://localhost:9080` on a Chromium browser: 
```
(index):5 Uncaught ReferenceError: require is not defined
    at (index):5:15
(anonymous) @ (index):5
(index):5 Uncaught ReferenceError: require is not defined
    at (index):5:124
(anonymous) @ (index):5
renderer.js:1 Uncaught ReferenceError: module is not defined
    at renderer.js:1:1
(anonymous) @ renderer.js:1
```

**Comment by k00l3** [comment]
I am having the same issue, downloaded the latest DMG and when I launch PS4 Remote PKG Sender V2.app  it just stays white as if it is frozen.  Intel MacOs 14.4

**Comment by Gkiokan** [comment]
You guys are awesome! Thanks for the reports. I will look this week into this issue.

**Comment by Gkiokan** [comment]
I upgraded my Mac OS version to 14.5 to see what really happens and I am shocked how much a breaking changes comes with the new OS build of mac. I wasn't expecting that and definetily not that much and deep.  This is what I call a good reason for "never change a running system". Sigh.  

So the Issue is that we have a couple OS wide function call deprecations for the CrashReporter, no problems here, we can omit it.  
DevTools aren't loading anymore due a not reported error. 
node-gyp does some crazy things on build and fails because of python issue on 3.10+.  
But the main problem is that the renderer can't use the nodeIntegration from chromium which leads finally to the white screen due the require is not a function error.  

But this is totally strange and makes no sense because the base version of the engine is still the same.  
I guess there are OS wide restrictions. I am still digging into it to find a workaround.  

Worst Case would be a rewrite of the engine with a bridged context which I am gonna avoid at all cost.  
I am on it. I will let you know. If anyone knows a fix for this bs*, feel free to comment.

**Comment by k00l3** [comment]
It's really weird it runs perfectly fine on my MacBook Pro M1  14.5 Beta (23F5049f) , Just having issues on my Intel based iMac 14.4

**Comment by Gkiokan** [comment]
Good sign for all! I got a live sign working now. Will continue tomorrow.
<img width="1116" alt="image" src="https://github.com/Gkiokan/ps4-remote-pkg-sender/assets/7249224/ab411d5b-1039-4284-9015-63d222de505c">

**Comment by Gkiokan** [comment]
https://github.com/Gkiokan/ps4-remote-pkg-sender/tree/macOSfixes

Here you can follow the branch for mac OS 14.4+ fixes.  
I've fixed the base dependencies, which results in getting the app back running.  
There are a couple things still to be fixed, but if anyone wants to test this (I don't have my ps4 connected rn), 
you can pull the branch and run a `npm i` and `npm run dev` to test the app.  

Once it's done it will be merged and released anyway.

**Comment by Gkiokan** [comment]
(v2.10.4)[https://github.com/Gkiokan/ps4-remote-pkg-sender/releases/tag/v2.10.4] just has been released and it should work properly again.  
I've tested it on my intel mac OS 14.5 which is currently the latest mac OS.  
 
I couldn't it on the Playstation though, but the manual endpoint tests where successful, so it must work.  
Have fun with the Hot fix Release. 

Side fixes: HB-Store Settings can be extended with a own custom HB-Store Server URL which can work in conjunction with the (HB-Store CDN Server)[https://github.com/Gkiokan/hb-store-cdn-server] or (HB-Store CDN CLI Server)[https://github.com/Gkiokan/hb-store-cdn-cli-server].
